cocos/scene-graph/node.ts

```diff
@@ -630,7 +630,7 @@ export class Node extends CCObject implements ISchedulable, CustomSerializable {
             return;
         }
         const siblings = this._parent._children;
-        index = index !== -1 ? index : siblings.length - 1;
+        index = index >= 0 ? index : siblings.length + index;
         const oldIndex = siblings.indexOf(this);
```

Comment:
Would it run well if passing a negative value that smaller than `siblings.length`?

Reply:
@kaokei 
Could you take a look at this question?
If `index` < `-siblings.length`, for example, it's `-siblings.length - 10`,
After this logic `index = index >= 0 ? index : siblings.length + index;`
The index will be `siblings.length + (-siblings.length - 10)` = `-10`, -10 will be a invalid index.

Reply:
It seems `splice` will check whether the index is in the range and do a clamp operation.